timer/api/client.go

```diff
@@ -200,9 +248,27 @@ func (c *defaultTimerClient) CloseTimerEvent(ctx context.Context, timerID string
 	if !update.Watermark.Present() {
 		update.Watermark.Set(timer.EventStart)
 	}
+	update.EventExtra.Set(EventExtra{})
 	return c.store.Update(ctx, timerID, update)
 }
 
 func (c *defaultTimerClient) DeleteTimer(ctx context.Context, timerID string) (bool, error) {
 	return c.store.Delete(ctx, timerID)
 }
+
+func (c *defaultTimerClient) retryWhenVersionNotMatch(ctx context.Context, timerID string, fn func() error) (err error) {
```

Comment:
Could you try to reuse the existing retry functions like [RunWithRetry](https://github.com/pingcap/tidb/blob/e76a3f363ab6335b333b8e89e1c95a7333a18696/util/misc.go#L66)?